src/main/java/picard/sam/SplitSamByNumberOfReads.java

```diff
@@ -106,11 +117,13 @@ protected int doWork() {
         final ProgressLogger firstPassProgress = new ProgressLogger(log, 1000000, "Counted");
         if (TOTAL_READS_IN_INPUT == 0) {
             final SamReader firstPass = readerFactory.referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
+            final SAMRecordIterator firstPassIterator = firstPass.iterator();
```

Comment:
sorry, I didn't realiz that you had two readers open already and my comments made you move from a for-loop to a while...I think that the original for loop is nicer.... :-/